### ISSUE TITLE
Better support for `Logic` based indexing of `List<Logic>`

### ISSUE THREAD

**Issue by mkorbel1** [intel/rohd#116]
## Motivation
In SystemVerilog, it's easy to index an entry in an array of signals:
```SystemVerilog
logic [8:0] a [4:0];
logic [2:0] b;
logic [8:0] c;

assign c = a[b];
```

ROHD doesn't have a synthesizable array type, instead encouraging usage of `List<Logic>`.  However, this makes it much more verbose to access an index of an array based on another `Logic`.

## Desired solution
Some sort of function or operation which reduces the effort required to do this sort of operation.

## Alternatives considered  
It is possible by looping through possible values manually and generating comparisons and `or`ing them together, but that's more verbose.

**Comment by mkorbel1** [comment]
Maybe a function on `Logic` called `select` or something that operates on an indexable `Iterable<Logic>` that automatically generates the hardware to select the right one?

**Comment by RPG-coder-intc** [comment]
Hey Max, I will start working on this one :)

**Comment by mkorbel1** [comment]
An example of an equivalent API in ROHD could be:
```dart
List<Logic> a; // length of list is 5, each element width is 9 bits
Logic b; // width is 3 bits
Logic c; // width is 9 bits

// two potential options:
c <= a.selectIndex(b); // order matches closer to array indexing
c <= b.selectFrom(a);
```

**Comment by RPG-coder-intc** [comment]
Working on this issue...

**Comment by mkorbel1** [comment]
Fixed in #438
